### DISCUSSION TITLE
Is it possible to inherit subfolder name as part of the trait value/name?

### DISCUSSION THREAD

**Discussion by echo-of-machines** [nftchef/art-engine#47] [Help + Q&A]
For example, if I have a folder named "Hair", in it is a bunch of subfolders like "Spikey", "Punk" etc, and within those are the colors, such as "Blue", "Red".  Is it possible to have the outcome of the trait type to be "Hair" while the trait value to be"Punk Red" which is the combination of the subfolder name and layer name?

**Comment by nftchef** [comment]
If I understand correctly, yes, 
take a look at config.js `useRootTraitType` and try setting it to `false`


<img width="575" alt="image" src="https://user-images.githubusercontent.com/91582112/155136931-cc38d5c8-e074-4f85-bad8-ba071f4316c0.png">

**Reply by echo-of-machines** [reply]
This would make the Trait Type "Spikey" or "Punk" while the value becomes "Red" or "Blue".  I still want the trait type to be "Hair", but the value to be "Punk Red"  which is a combination of subfolder and the png name.  Is this possible?

**Reply by nftchef** [reply]
The generator only uses the file or folder name for the trait, not a combination as that can have side effects.
You _could_ modify the code to do this, but the recommended way would be to rename the file as the desired trait name. `Punk Red#5.png` for example

**Comment by soulmv** [comment]
What if you create another folder "Hair Color" and put transparent layers with color filenames there and bind them to each color in "Hair folder? So there will be two traits in metadata - Hair and Hair Color

Hair
-- Spikey#100
    --- Spikey Blue#50.png
    --- Spikey Red#50.png

Hair Color
-- Blue#50.png
-- Red#50.png



```
const forcedCombinations = {
"Spikey Blue": ["Blue"],
"Spikey Red": ["Red"],
};
```

```
const traitValueOverrides = {
"Spikey Blue" : "Spikey",
"Spikey Red" : "Spikey",
};
```
